libarchive/archive_write.c

```diff
@@ -354,6 +354,10 @@ archive_write_client_open(struct archive_write_filter *f)
 	int ret;
 
 	f->bytes_per_block = archive_write_get_bytes_per_block(f->archive);
+	if (f->bytes_per_block < 0){
+                // The `__archive_check_magic` function already set an error
+        	return (ARCHIVE_FATAL);
+        }
 	f->bytes_in_last_block =
 	    archive_write_get_bytes_in_last_block(f->archive);
 	buffer_size = f->bytes_per_block;
```

Comment:
There are two interesting questions here:
* Does this code always work correctly as long as `bytes_per_block` is at least one?   I believe it should; I know we did a lot of work on archive_read to make sure it always worked correctly even with one-byte blocks.
* Can `bytes_per_block` ever be set to zero or less?   It's easy for `set_bytes_per_block` to refuse to set a value less than one.